### ISSUE TITLE
Error: Not implemented Error: Expr of type 347 not implemented

### ISSUE THREAD

**Issue by nelhage** [duckdb/duckdb#1619]
Given a Parquet file containing a nested struct, (using the same file from #1618  -- https://github.com/nelhage/duckdb-parquet-bugs/blob/master/data/struct.parquet), I can read fields using `struct_extract`:

```
D select struct_extract("inner", 'f64_field') from parquet_scan('data/struct.parquet');
struct_extract(inner, f64_field)
--------------------------------
1.23

```

But attempting to read fields using `struct[field]`  -- [as documented here](https://duckdb.org/docs/sql/functions/nested) -- errors:

```
D select "inner"['f64_field'] from parquet_scan('data/struct.parquet');
Error: Not implemented Error: Expr of type 347 not implemented

```

**Comment by Mytherin** [comment]
Thanks for the report! The documentation is a bit ahead of the current release. On the master branch this code reports the following error:

```sql
Binder Error: No function matches the given name and argument types 'array_extract(STRUCT<str_field: VARCHAR, f64_field: DOUBLE>, VARCHAR)'. You might need to add explicit type casts.
	Candidate functions:
	array_extract(ANY, BIGINT) -> ANY
```

That is because the alias for struct_extract is a dot, e.g.:

```sql
select ("inner").'f64_field' from parquet_scan('data/struct.parquet');
```

(currently quotes are required). However, I think supporting the struct['field'] syntax is a good idea. I have pushed support for this syntax in #1621.

**Comment by hannesmuehleisen** [comment]
Should be fixed through #1621
